test/over_react/component/shared_error_boundary_tests.dart

```diff
@@ -164,16 +184,20 @@ void sharedErrorBoundaryTests(BuilderOnlyUiFactory builder) {
           final component = jacket.getDartInstance();
           component.setState(component.newState()
             ..hasError = true
-            ..showFallbackUIOnError = true
           );
           expect(getByTestId(jacket.getInstance(), 'dummyChild'), isNull);
-          expect(getByTestId(jacket.getInstance(), 'fallbackNode'), isNotNull);
+          if (!isWrapper) {
+            // wrapper ErrorBoundary doesn't use `props.fallbackUIRenderer` because `ResolvableErrorBoundary`
+            // will display it on first error. Meaning wrapper ErrorBoundary will never be reached if it is set.
+            expect(getByTestId(jacket.getInstance(), 'fallbackNode'), isNotNull);
+          }
         });
 
         test('when reset() is called', () {
-          jacket.getDartInstance().reset();
-          expect(jacket.getDartInstance().state.hasError, isFalse);
-          expect(jacket.getDartInstance().state.showFallbackUIOnError, isTrue);
+          dynamic component = jacket.getDartInstance();
+          (component as dynamic).reset();
```

Comment:
I don't think making this dynamic is necessary.  You just need to update line 15 to:

```dart
TestJacket<ErrorBoundaryMixin> jacket; // ignore: deprecated_member_use_from_same_package
```

and then this can change to:

```dart
final component = jacket.getDartInstance();
component.reset();
```

Reply:
i cannot because ErrorBoundary no longer has anything to do with ErrorBoundaryMixin.

Reply:
k… then lets moved the shared public method in an interface like so:

mixin ErrorBoundaryApi on UiStatefulComponent2 {
  /// Resets the [ErrorBoundary] to a non-error state.
  ///
  /// This can be called manually on the component instance using a `ref` -
  /// or by passing in a new child instance after a child has thrown an error.
  void reset() {
    setState(initialState);
  }
}

The interface should not be exported publicly, but it can then be mixed in to both of the error boundary components.